x/evm/wrappers/bank.go

```diff
@@ -0,0 +1,111 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package wrappers
+
+import (
+	"context"
+	"math/big"
+
+	"cosmossdk.io/errors"
+	sdkmath "cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/evmos/evmos/v20/x/evm/config"
+	"github.com/evmos/evmos/v20/x/evm/types"
+)
+
+var _ types.BankWrapper = BankWrapper{}
+
+// BankWrapper is a wrapper around the Cosmos SDK bank keeper
+// that is used to manage an evm denom with a custom decimal representation.
+type BankWrapper struct {
+	types.BankKeeper
+}
+
+// NewBankWrapper creates a new BankWrapper instance.
+func NewBankWrapper(
+	bk types.BankKeeper,
+) *BankWrapper {
+	return &BankWrapper{
+		bk,
+	}
+}
+
+// ------------------------------------------------------------------------------------------
+// Bank wrapper own methods
+// ------------------------------------------------------------------------------------------
+
+// MintAmountToAccount converts the given amount into the evm coin scaling
+// the amount to the original decimals, then mints that amount to the provided account.
+func (w BankWrapper) MintAmountToAccount(ctx context.Context, recipientAddr sdk.AccAddress, amt *big.Int) error {
+	coin := sdk.NewCoin(config.GetEVMCoinDenom(), sdkmath.NewIntFromBigInt(amt))
+
+	convertedCoin, err := convertEvmCoinFrom18Decimals(coin)
```

Comment:
nit: convert directly without checking if the denom is the evm denom (check in `convertEvmCoinFrom18Decimals`)

Reply:
The conversion is a very critical part for the evm and the bank, I believe that until the entire 6 decimal is ready and we confirm it is working as expected, we should keep this safety measure. It is just an equality check of short strings so not very expensive.